src/Microsoft.Diagnostics.Tools.RuntimeClient/Eventing/EventPipeClient.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+using System.Security.Principal;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Diagnostics.Tools.RuntimeClient.Eventing
+{
+    public static class EventPipeClient
+    {
+        // TODO: Make async.
```

Comment:
I'd actually suggest not doing that. Async helps for scaling to high numbers of concurrent IO operations, but I'd expect typical concurrency will be 1-5 parallel operations in this library and it would be surprising if anyone ever exceeded 50. Dedicated threads can easily handle that scale, no extra work is involved, and the extra complexity of async is avoided.